storage/innobase/xtrabackup/doc/source/xbcloud/xbcloud.rst

```diff
@@ -28,40 +35,237 @@ Version specific information
+.. rubric:: Swift
+
+The following example shows how to make a full backup and upload it to Swift
 
 .. code-block:: bash
 
    $ xtrabackup --backup --stream=xbstream --extra-lsndir=/tmp --target-dir=/tmp | \
-   xbcloud put --storage=Swift \
+   xbcloud put --storage=swift \
    --swift-container=test \
    --swift-user=test:tester \
    --swift-auth-url=http://192.168.8.80:8080/ \
    --swift-key=testing \
    --parallel=10 \
    full_backup
 
-.. rubric:: Restore:
+.. rubric:: Amazon S3
+
+.. code-block:: bash
+
+   $ xtrabackup --backup --stream=xbstream --extra-lsndir=/tmp --target-dir=/tmp | \
+   xbcloud put --storage=s3 \
+   --s3-endpoint='s3.amazonaws.com' \
+   --s3-access-key='YOUR-ACCESSKEYID' \
+   --s3-secret-key='YOUR-SECRETACCESSKEY' \
+   --s3-bucket='mysql_backups'
+   --parallel=10 \
+   ${date}-full_backup
+
+The following options are available when using Amazon S3:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Option
+     - Details
+   * - --s3-access-key
+     - Use to supply the AWS access key ID
+   * - --s3-secret-key
+     - Use to supply the AWS secret access key
+   * - --s3-bucket
+     - Use supply the AWS bucket name
+   * - --s3-region
+     - Use to specify the AWS region. The default value is **us-east-1**
+   * - --s3-api-version = <AUTO|2|4>
+     - Select the signing algorithm. The default value is AUTO. In this case, |xbcloud| will probe.
+   * - --s3-bucket-lookup = <AUTO|PATH|DNS>
+     - Specify whether to use **bucket.endpoint.com** or *endpoint.com/bucket**
+       style requests. The default value is AUTO. In this case, |xbcloud| will probe.
+
+The following environment variables are recognized:
+
+.. rubric:: MinIO
+	    
+.. code-block:: bash
+
+   $ xtrabackup --backup --stream=xbstream --extra-lsndir=/tmp --target-dir=/tmp | \
+   xbcloud put --storage=s3 \
+   --s3-endpoint='play.minio.io:9000' \
+   --s3-access-key='YOUR-ACCESSKEYID' \
+   --s3-secret-key='YOUR-SECRETACCESSKEY' \
+   --s3-bucket='mysql_backups'
+   --parallel=10 \
+   ${date}-full_backup
+
+.. rubric:: |gcs|
+
+The support for |gcs| is implemented using the interoperability
+mode. This mode was especially designed to interact with cloud services
+compatible with Amazon S3.
+
+.. seealso::
+
+   Cloud Storage Interoperability
+      https://cloud.google.com/storage/docs/interoperability
+
+.. code-block:: bash
+		
+   $ xtrabackup --backup --stream=xbstream --extra-lsndir=/tmp --target-dir=/tmp | \
+   xbcloud put --storage=google \
+   --google-endpoint=`storage.googleapis.com` \
+   --google-access-key='YOUR-ACCESSKEYID' \
+   --google-secret-key='YOUR-SECRETACCESSKEY' \
+   --google-bucket='mysql_backups'
+   --parallel=10 \
+   ${date}-full_backup
+
+The following options are available when using |gcs|:
+
+.. hlist::
+   :columns: 2
+
+   - --google-access-key = <ACCESS KEY ID>
+   - --google-secret-key = <SECRET ACCESS KEY>
+   - --google-bucket = <BUCKET NAME>
+   - --google-region = <REGION> (default is us-central-1)
```

Comment:
This option has no effect and will be removed. Lets not include it into the doc

Reply:
https://jira.percona.com/browse/PXB-1848